packages/react-components/react-input/src/components/Input/Input.types.ts

```diff
@@ -58,6 +58,13 @@ export type InputProps = Omit<
     | 'filled-darker-shadow'
     | 'filled-lighter-shadow';
 
+  /**
+   * Causes the border to be red, indicating that the value entered by the user has failed validation.
+   *
+   * It is recommended to set `aria-invalid` and `aria-errormessage` as well.
+   */
+  invalid?: boolean;
```

Comment:
One problem with a new `invalid` prop is that when used with `InputField`, it gives you two ways to say the same thing: `<InputField invalid />` is the same as `<InputField validationState="error" />`. 

I considered a few possible alternative approaches, and wanted to see if people preferred any of them:
1. Instead of an `invalid` prop, add a `validationState?: 'success' | 'warning' | 'error'` prop to Input, which matches Field's prop. The main drawback here is that `<Input validationState="success" />` (or "warning") doesn't do anything, but all values need to be allowed in the types in order to work with Field.
2. Don't add any new prop to `Input` and instead have the border be red when `aria-invalid` is true (which gets set by Field in this case). The main drawback is that there's then no way to have a red border without `aria-invalid`, and no way to have `aria-invalid` without a red border.